packages/adapter/addon/rest.ts

```diff
@@ -1146,12 +1145,14 @@ class RESTAdapter extends Adapter.extend(BuildURLMixin) {
       options = ajaxOptions(options, this);
     }
 
-    options.url = this._ajaxURL(options.url);
+    if (options.url) {
```

Comment:
Do you think this if changes things?

Reply:
Ok fixed this!  I created a new variable `reqOptions` with our own "options" type extending the JQuery and Fetch Init typings.

https://github.com/emberjs/data/pull/7358/commits/6a9bf658a51c8eb5338a7ae0f798f21068713278